src/plans/plan-interface.h

```diff
@@ -89,20 +44,29 @@ LibraryAction DeliverSample (In Real X,
                              In Real Y,
                              In Real Z);
 
-// Does nothing
-LibraryAction Stub (In String desc);
-
 
-// Lookups
-
-// Can be used for fine-grain process control, but generally not needed.
-Boolean Lookup Running (String operation_name);
-Boolean Lookup Finished (String operation_name);
+// Lander queries
 
 Boolean Lookup HardTorqueLimitReached (String joint_name);
 Boolean Lookup SoftTorqueLimitReached (String joint_name);
+Boolean Lookup ImageReceived;
+
+// Relevant with GuardedMove only:
 Boolean Lookup GroundFound;
 Real    Lookup GroundPosition;
-Boolean Lookup ImageReceived;
+
+
+// Misc
+
+// Query whether a given operation has started, or is finished after having been
+// started.  These use the operation names as defined in OwInterface.cpp, and
+// offer fine-grain process control.  They are not required for general use.
+Boolean Lookup Running (String operation_name);
+Boolean Lookup Finished (String operation_name);
```

Comment:
Probably outside the scope for this ticket but I think this query is actually misleading! I do recommend removing it.

Reply:
Yes, outside scope of this ticket.  These are now used only by a couple of test plans, and are intended for concurrency control from PLEXIL, which in general should not be needed now that Services are threaded and Actions are on the way.  

I can now see that Finished is misleading and should be removed -- it is clearer to query !Running.  If you see another issue, please file a Jira.

Reply:
Exactly, the term **Finished** gives you the impression the operation ran and completed (regardless of outcome). But you would still get **Finished** even when the operation was never started.